### PR TITLE
Full beam search export via Pytorch hybrid frontend and ONNX

### PR DESCRIPTION
**Pull request by jamesr66a**

### CODE REVIEW

File: pytorch_translate/ensemble_export.py [hunk 1]
```diff
@@ -818,3 +769,154 @@ def save_to_db(self, output_path, encoder_ensemble_outputs):
...
+
+class BeamSearch(torch.jit.ScriptModule):
+
+    __constants__ = ['bs']
+
+    def __init__(self, model_list, src_tokens, src_lengths, beam_size=1,
+                 word_penalty=0, unk_penalty=0):
+        super().__init__()
+        self.models = model_list
+        self.bs = beam_size
```
[the remark this answers is not in the document]
Comment (jamesr66a) [reply pytorch_translate/ensemble_export.py]: It makes it really long in the forward method and makes it so i have to do ugly re-flowing of the lines there

[the remark this answers is not in the document]
Comment (theweiho) [reply pytorch_translate/ensemble_export.py]: Can you just multi-line the forward method? I think having sensible variable names is more important for code readability

**Comment by theweiho** [comment]
Thanks! I think we should make sure the example scripts to export and load the models under https://github.com/pytorch/translate/tree/master/pytorch_translate/examples remain working in conjunction

**Comment by theweiho** [comment]
@jmp84 - doesn't this make our example scripts to export and load the models no longer work?

**Comment by theweiho** [comment]
Ah sorry - I think I misunderstood. The original export method is still kept, and not replaced in this PR

**Pull request merged**